packages/@uppy/aws-s3/src/MultipartUploader.ts

```diff
@@ -236,7 +236,9 @@ class MultipartUploader<M extends Meta, B extends Body> {
       this.#resumeUpload()
     } else if (this.#isRestoring) {
       this.options.companionComm.restoreUploadFile(this.#file, {
-        uploadId: this.options.uploadId,
+        ...(this.options.uploadId !== undefined && {
```

Comment:
here i agree we should use the `!== undefined check` because the string could be `''` which we probably want to include